onnxruntime/core/providers/cpu/quantization/quantize_linear.cc

```diff
@@ -132,9 +212,41 @@ Status DequantizeLinear<T>::Compute(OpKernelContext* ctx) const {
 
 REGISTER_QUANTIZELINEAR(int8_t)
 REGISTER_QUANTIZELINEAR(uint8_t)
+
+REGISTER_QUANTIZELINEAR(Float8E4M3FN)
+REGISTER_QUANTIZELINEAR(Float8E4M3FNUZ)
+REGISTER_QUANTIZELINEAR(Float8E5M2)
+REGISTER_QUANTIZELINEAR(Float8E5M2FNUZ)
+
 REGISTER_QUANTIZELINEAR_VERSIONED(int8_t)
 REGISTER_QUANTIZELINEAR_VERSIONED(uint8_t)
 
+template <typename OutputType>
+typename std::enable_if<!boost::mp11::mp_contains<element_type_lists::AllFloat8, OutputType>::value, void>::type
+ParQuantizeLinear(const float* Input,
+                  OutputType* Output,
+                  size_t N,
+                  float Scale,
+                  size_t bd,
+                  const OutputType* ZeroPoint,
+                  bool /* saturate */,
+                  concurrency::ThreadPool* thread_pool) {
+  ParQuantizeLinearStd(Input, Output, N, Scale, ZeroPoint != nullptr ? ZeroPoint[bd] : (OutputType)0, thread_pool);
+}
+
+template <typename OutputType>
+typename std::enable_if<boost::mp11::mp_contains<element_type_lists::AllFloat8, OutputType>::value, void>::type
+ParQuantizeLinear(const float* Input,
+                  OutputType* Output,
+                  size_t N,
+                  float Scale,
+                  size_t bd,
+                  const OutputType* ZeroPoint,
+                  bool saturate,
+                  concurrency::ThreadPool* thread_pool) {
+  ParQuantizeLinearSat(Input, Output, N, Scale, ZeroPoint != nullptr ? ZeroPoint[bd] : OutputType(static_cast<float>(0), true), saturate, thread_pool);
+}
```

Comment:
simplify with if constexpr
```suggestion
template <typename OutputType>
void ParQuantizeLinear(const float* Input,
                       OutputType* Output,
                       size_t N,
                       float Scale,
                       size_t bd,
                       const OutputType* ZeroPoint,
                       bool saturate,
                       concurrency::ThreadPool* thread_pool) {
  if constexpr (!boost::mp11::mp_contains<element_type_lists::AllFloat8, OutputType>::value) {
    ORT_UNUSED_PARAMETER(saturate);
    ParQuantizeLinearStd(Input, Output, N, Scale, ZeroPoint != nullptr ? ZeroPoint[bd] : (OutputType)0, thread_pool);
  } else {
    ParQuantizeLinearSat(Input, Output, N, Scale, ZeroPoint != nullptr ? ZeroPoint[bd] : OutputType(static_cast<float>(0), true), saturate, thread_pool);
  }
}
```